### ISSUE TITLE
Audit error: Required Accessibility gatherer encountered an error when using almond.js

### ISSUE THREAD

**Issue by karl-sjogren** [GoogleChrome/lighthouse#2505]
We are using https://github.com/requirejs/almond in our builds instead of the whole requirejs package and this seems to break lighthouse accessibility tests. I've tried it both via the node package and via the Audit tab in Chrome Developer Tools.

    lighthouse https://bibliotek.jonkoping.se/

The result is a 0 score for accessibility and all tests have failed with the below message:

    Audit error: Required Accessibility gatherer encountered an error: See almond README: incorrect module build, no module name

Almond has 30 000 downloads the last week on npm and is used in lots of places around the web so a solution or at least a workaround that we can use would be great!

**Comment by patrickhulce** [comment]
This seems to be an issue with [axe](https://github.com/dequelabs/axe-core) interacting with your site. Would you mind filing this issue upstream?

I get this stack trace when just trying to run axe on your site without Lighthouse
```
Uncaught Error: See almond README: incorrect module build, no module name
    at define (vendor-9f0474e5e4.js.br:1)
    at axeFunction (axe.js:24)
    at axe.js:8472
```

**Comment by mjpuser** [comment]
I am seeing a different error, but accessibility tests are all broken:

```bash
$ lighthouse --version
2.1.0
```
```bash
Runner:warn Accessibility gatherer, required by audit video-description, encountered an error:
Cannot read property 'push' of undefined +0ms
video-description:warn Caught exception: Required Accessibility gatherer encountered an error:
Cannot read property 'push' of undefined +0ms
```

Notice they all say `Cannot read property 'push' of undefined`

**Comment by ebidel** [comment]
@mjpuser can you post the URL of your site?

**Comment by ebidel** [comment]
> aweber.com

**Comment by karl-sjogren** [comment]
Sorry for the late feedback here but it's easy to forget stuff after a few weeks of vacation.

The thing is that it works fine with their own Developer Tools plugin so I feel that it might be up to how you invoke it. I've solved this on my end using an IIFE around my code anyway so if anyone run into this on their own site then this is a possible workaround.

**Comment by paulirish** [comment]
Ok. closing this for now as it's resolved in your case. 

If someone else runs into this we'll have to investigate if this is a bug with AXE's UMD or how we're running it.

**Comment by shellbryson** [comment]
First time looking at the new auditing on a reasonably sized project and seeing the issue above. Uses RequireJS, but the dev tools appear to choke:

![screen shot 2017-08-30 at 15 11 49](https://user-images.githubusercontent.com/1268036/29876845-bde4554c-8d95-11e7-8ef1-fda5caf5dac2.png)

**Comment by shellbryson** [comment]
https://www.scottishfriendly.co.uk/

**Comment by patrickhulce** [comment]
If it works with axe's extension, it's likely because they're running it inside a content script that won't be affected by the globals of the page. Migrating our page evaluations to `createIsolatedWorld` should fix this for us.

**Comment by paulirish** [comment]
in that case, This is blocked by #2505
